src/main/java/net/sf/jabref/cli/ArgumentProcessor.java

```diff
@@ -0,0 +1,548 @@
+import net.sf.jabref.importer.ImportFormatReader;
+import net.sf.jabref.importer.ImportInspectionCommandLine;
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fetcher.EntryFetcher;
+import net.sf.jabref.importer.fetcher.EntryFetchers;
+import net.sf.jabref.logic.CustomEntryTypesManager;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.labelpattern.LabelPatternUtil;
+import net.sf.jabref.logic.logging.JabRefLogger;
+import net.sf.jabref.logic.search.DatabaseSearcher;
+import net.sf.jabref.logic.search.SearchQuery;
+import net.sf.jabref.logic.util.OS;
+import net.sf.jabref.logic.util.strings.StringUtil;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.database.BibDatabaseMode;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.util.Util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class ArgumentProcessor {
+
+    public enum Mode {
+        INITIAL_START,
+        REMOTE_START
+    }
+
+
+    private static final Log LOGGER = LogFactory.getLog(ArgumentProcessor.class);
+
+    private final JabRefCLI cli;
+
+    private final Optional<List<ParserResult>> parserResults;
+
+    private final Mode startupMode;
+
+
+    public ArgumentProcessor(String[] args, Mode startupMode) {
+        cli = new JabRefCLI(args);
+        this.startupMode = startupMode;
+        parserResults = processArguments();
+    }
+
+    public Optional<List<ParserResult>> getParserResults() {
+        return parserResults;
+    }
+
+    public boolean hasParserResults() {
+        return parserResults.isPresent();
+    }
+
+    private Optional<List<ParserResult>> processArguments() {
+
+        if (!cli.isBlank() && cli.isDebugLogging()) {
+            JabRefLogger.setDebug();
+        }
+
+        if ((startupMode == Mode.INITIAL_START) && cli.isShowVersion()) {
+            cli.displayVersion();
+        }
+
+        if ((startupMode == Mode.INITIAL_START) && cli.isHelp()) {
+            cli.printUsage();
+            return Optional.empty();
+        }
+
+        // Check if we should reset all preferences to default values:
+        if (cli.isPreferencesReset()) {
+            resetPreferences(cli.getPreferencesReset());
+        }
+
+        // Check if we should import preferences from a file:
+        if (cli.isPreferencesImport()) {
+            importPreferences();
+        }
+
+        // List to put imported/loaded database(s) in.
+        List<ParserResult> loaded = new ArrayList<>();
+        importAndOpenFiles(loaded);
+
+        if (!cli.isBlank() && cli.isFetcherEngine()) {
+            fetch(cli.getFetcherEngine()).ifPresent(loaded::add);
+        }
+
+        if (cli.isExportMatches()) {
+            if (!loaded.isEmpty()) {
+                String[] data = cli.getExportMatches().split(",");
+                String searchTerm = data[0].replace("\\$", " "); //enables blanks within the search term:
```

Comment:
Returning a boolean which indicates whether the call was successful or not does not work?